code/modules/hippie/code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -0,0 +1,25 @@
+// -------------------------------------
+// Kitchen Smartfridge
+// -------------------------------------
+/obj/machinery/smartfridge/kitchen
+	name = "harvest storage"
+	desc = "Holds and preserves organically grown biomasses. Otherwise known as plants."
+	var/list/spawn_harvest = list(
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = 3,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/ambrosia/vulgaris = 3,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/eggplant = 2,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/carrot = 2,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/corn = 2,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/apple = 2,
+		/obj/item/weapon/reagent_containers/food/snacks/grown/wheat = 2)
+
+
+/obj/machinery/smartfridge/kitchen/New()
+	..()
+	for(var/typekey in spawn_harvest)
+		var/amount = spawn_harvest[typekey]
+		if(isnull(amount)) amount = 1
+		while(amount)
```

Comment:
Do negative values return true or false when comparing them? If it's false then no need to change. My concern was -1 would be "true" (as it's non-zero value) so if somehow you had amount = -1 you could potentially have infinite loop